### PR TITLE
Update testing docs with info about React Testing Library

### PR DESCRIPTION
**Pull request by jeyip**
## Description
After discussions with @nerrad, the team is favoring React Testing Library over Enzyme for React component testing. This, however, is not reflected in the documentation. We update the `testing-overview` markdown file to include this information.

## How has this been tested?
N/A

## Screenshots 
N/A

## Types of changes
Documentation changes

## Checklist:
N/A

### CODE REVIEW

File: docs/contributors/testing-overview.md [hunk 1]
```diff
@@ -19,7 +19,9 @@ When writing tests consider the following:
...
+Tests for JavaScript use [Jest](https://jestjs.io/) as the test runner and its API for [globals](https://jestjs.io/docs/en/api.html) (`describe`, `test`, `beforeEach` and so on) [assertions](https://jestjs.io/docs/en/expect.html), [mocks](https://jestjs.io/docs/en/mock-functions.html), [spies](https://jestjs.io/docs/en/jest-object.html#jestspyonobject-methodname) and [mock functions](https://jestjs.io/docs/en/mock-function-api.html). If needed, you can also use [React Testing Library](https://testing-library.com/docs/react-testing-library/intro) for React component testing.
+
+It should be noted that in the past, React components were unit tested with [Enzyme](https://github.com/airbnb/enzyme). We are, however, slowly moving development away from that library on an ongoing basis.
```
Comment (nerrad) [inline docs/contributors/testing-overview.md:24]:
```suggestion
It should be noted that in the past, React components were unit tested with [Enzyme](https://github.com/airbnb/enzyme). However, for new tests, it is preferred to use React Testing Library (RTL) and over time old tests should be refactored to use RTL too (typically when working on code that touches an old test).
```

**Review by nerrad — COMMENTED** [review]
Enzyme is also being used for one of the examples in the section talking about snapshot testing. We'll want to replace that example at some point too.

However, I think this tweak in documentation will at least help to prevent some confusion around what library to prefer for writing tests.

Thanks for adding this @jeyip! I have one suggested change.

**Review by jeyip — COMMENTED** [review]
@nerrad Awesome! I accepted the suggested change.

Do you happen to know where we could leave a note to return to this markdown document and update the unit test examples with RTL? Would it be an issue filed with Gutenberg?

**Comment by gziolo** [comment]
> Enzyme is also being used for one of the examples in the section talking about snapshot testing. We'll want to replace that example at some point too.

Do we have any test that could be used instead there?

**Comment by nerrad** [comment]
> Do you happen to know where we could leave a note to return to this markdown document and update the unit test examples with RTL? Would it be an issue filed with Gutenberg?

I would say an issue could be fine, _ideally_ we'd do it as a part of this pull.

> Do we have any test that could be used instead there?

Ya that'd be a good idea, @jeyip you could likely just look for an existing rtl test in the repo (maybe [one of these results](https://github.com/WordPress/gutenberg/search?q=testing-library%2Freact&unscoped_q=testing-library%2Freact) would be good) and use it?

**Review by jeyip — COMMENTED** [review]
> Ya that'd be a good idea, @jeyip you could likely just look for an existing rtl test in the repo (maybe one of these results would be good) and use it?

@nerrad Sure. I found a few examples and I'll update this PR when I get a chance.

**Review by gziolo — APPROVED** [review]
Let’s merge it as is. We can always update examples separately 😃

**Pull request merged**